workshops/geometric_pattern/README.md

```diff
@@ -0,0 +1,538 @@
+
+You can play around with these values to get a better feel of how this function works.
+
+NOTE TO SELF: is this necessary. should we just be focusing on the pattern instead of exploring p5.js
+
+### Drawing a Row of Circles
+
+Since we want to draw circles repeatedly, we have to write `ellipse()` as many times as we want circles.
+
+Just kidding! There's a construct in programming called a for-loop, and it repeats a set of instructions as many times as you decide.
+
+Let's remove our previous line and add a for-loop into our `draw()` function, like so:
+
+```js
+function draw() {
+  for (var x = 0; x <= width; x = x+CIRCLE_DIA) {
+  }
+}
+```
+
+This for-loop executes the code within it over the period when the value of `x`, which we've initialized at `0`, is less than or equal to `width`. In each iteration of the loop, the value of `x` is increased by `CIRCLE_DIA`.
+
+If we save and refresh, we'll see nothing. That's because we there is no code inside the for-loop. Let's add a line that draws ellipses along a row at the top of the canvas:
+
+```js
+function draw() {
+  for (var x = 0; x <= width; x = x+CIRCLE_DIA) {
+    ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+  }
+}
+```
+
+We're supplying `x` as the x-coordinate (cleverly named, eh?), and 0 as the y-coordinate, of the ellipse's center.
+
+If you save and refresh, you'll see a line of cut-off circles at the top. This is the magic of the for-loop. Since the value of `x` was increased by `CIRCLE_DIA` during every repeated call of the `ellipse()` function, circles were drawn in intervals of `CIRCLE_DIA` pixels.
+
+As to why the circles are cut off -- this is because we set the y-coordinate of the _center_ of every circle to 0.
+
+### Drawing a Grid of Circles
+
+So that's great, we've drawn one row of circles. But what we'd like is to cover the entire canvas with circles.
+
+Just like we used a for-loop to repeat circles in the horizontal direction, we can also use a for-loop to repeat them in the vertical direction.
+
+If we wrap our existing for-loop in another for-loop, we'll be performing the action of filling an entire row with circles, multiple times. And thus covering multiple rows.
+
+Let's put everything so far inside another for-loop:
+
+```js
+function draw() {
+  for (var y = 0; y <= height; y = y+CIRCLE_DIA) {
+    for (var x = 0; x <= width; x = x+CIRCLE_DIA) {
+      ellipse(x, 0, CIRCLE_DIA, CIRCLE_DIA);
+    }
+  }
+}
+```
+
+As with the previous for-loop's `x`, we're executing code as long as the value of `y`, which starts at `0`, is less than or equal to `height`. (`height`, as you may have guessed, is the counterpart to `width`, and it is where p5.js stores the height of the canvas.) We're also incrementing `y` at each iteration by `CIRCLE_DIA`.
+
+Save and refresh Live Preview to check it out!
```

Comment:
I actually think that loops inside loops is made very clear thanks to the Live Preview. This also wasn't a problem in testing.